src/main/java/application_business_rules/ManagementSystem.java

```diff
@@ -118,7 +118,7 @@ public boolean presNameChecker(String presName){
     }
 
     public void addNewPrescription(List<String> medicines, String presName){
-        List<Medicine> allMedicines = List.copyOf(userManager.getMedicines().values());
+        List<Medicine> allMedicines = List.copyOf(userManager.getUser().getMedicineList().values());
```

Comment:
We could perhaps change this so that we don't reveal the representation of the implementation of UserManager to ManagementSystem. We can do this by adding the method getMedicines in UserManager.